tools/build_newlib.sh

```diff
@@ -0,0 +1,91 @@
+#!/usr/bin/env bash
+
+set -exu
+
+TARGET_CPU=""
+OUTPUT_DIR=""
+
+while getopts "t:o:" opt
+do
+    case "$opt" in
+        t)
+            TARGET_CPU="$OPTARG"
+            ;;
+        o)
+            OUTPUT_DIR="$OPTARG"
+            ;;
+        ?)
+            exit 1
+            ;;
+    esac
+done
+shift "$((OPTIND - 1))"
+
+if [ -z "${TARGET_CPU}" ] || [ -z "${OUTPUT_DIR}" ]
+then
+    echo "Usage: $0 -t TARGET_CPU -o OUTPUT_FILE" >&2
+    exit 1
+fi
+
+
+mkdir -p "$OUTPUT_DIR"
+OUTPUT_DIR=$(realpath "$OUTPUT_DIR")
+
+
+# ENV SETUP
+WORKDIR="$(pwd)"
+SRC="${WORKDIR}/newlib"
+BUILD="${SRC}/arm_none_eabi_build"
+INSTALL="${SRC}/arm_none_eabi_install"
+cd "${WORKDIR}"
+
+# Installing newlib build dependencies
+apt update
+apt install -y --no-install-recommends texinfo
+
+# Get the latest fixed version
+git clone --depth=1 --single-branch --branch newlib-4.5.0 https://sourceware.org/git/newlib-cygwin.git "${SRC}"
```

Comment:
I would prefer if the newlib sources were downloaded with `apt source` like how it is done for LLVM in the other script.
That way it would have the same exact version as the one already installed.
Because like this, an app build would use the newlib 4.5.0 static libraries but the 3.3.0 header files.
If newlib 4.5.0 is somehow required, maybe we should wait for the migration of the app-builder container to Debian 13.